### ISSUE TITLE
Shellcheck test report

### ISSUE THREAD

**Issue by ghost** [slimm609/checksec.sh#49]
Hi!
There is a nice tool called [Shellcheck](https://github.com/koalaman/shellcheck) to statically analyze shell scripts. I performed simple test and it shows many potential issues. If you are interested you can look at the report or perform test yourself.
https://www.shellcheck.net/
[shellecheck_test.txt](https://github.com/slimm609/checksec.sh/files/1188622/shellecheck_test.txt)

**Comment by NellePfannerstill72** [comment]
I fixed all quotes so shellcheck report should be more readable now. I leave the rest for your judgment.
[shellcheck.txt](https://github.com/slimm609/checksec.sh/files/1206360/shellcheck.txt)

**Comment by NellePfannerstill72** [comment]
Great work! That fixed some bogus errors while reading selinux with `checksec -k` status on my machine. There are still some after executing `checksec -pa` with some binaries:
```
readelf: Warning: local symbol 1 found at index >= .dynsym's sh_info value of 1
readelf: Warning: local symbol 2 found at index >= .dynsym's sh_info value of 1
readelf: Warning: local symbol 3 found at index >= .dynsym's sh_info value of 1
```
Above comes after reading fortify status on rtkit-daemon. Nothing important to me, you can ignore if it's not easily fixable.


That's what left in shellcheck report after your recent changes:
```
In /usr/bin/checksec line 551:
  $debug && echo "$($kconfig | grep "CONFIG_GRKERNSEC")"
                 ^-- SC2005: Useless echo? Instead of 'echo $(cmd)', just use 'cmd'.


In /usr/bin/checksec line 552:
  $debug && echo "$($kconfig | grep "CONFIG_PAX")"
                 ^-- SC2005: Useless echo? Instead of 'echo $(cmd)', just use 'cmd'.


In /usr/bin/checksec line 1133:
  if ! $(openssl dgst -sha256 -verify "${PUBKEY_FILE}" -signature "${SIG_FILE}" "${TMP_FILE}" >/dev/null 2>&1); then
       ^-- SC2091: Remove surrounding $() to avoid executing output.


In /usr/bin/checksec line 1143:
    if [[ $? == 0 ]]; then
          ^-- SC2181: Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.


In /usr/bin/checksec line 1342:
    for N in $(ps -Ao pid,comm | grep "$2" | cut -b1-6); do
               ^-- SC2009: Consider using pgrep instead of grepping ps output.


In /usr/bin/checksec line 1534:
      echo_message "" "" "<fortify-test name='$name' pid='$N' " "{ \"fortify-test\": { \"name\":\"$name\", \"pid\":\"$pid\", "
                                                                                                                     ^-- SC2154: pid is referenced but not assigned.
```
If you think above is normal, please close this issue.

**Comment by slimm609** [comment]
i got it down to 2, which i am just going to leave until i get some additional time to test and update them.

**Comment by NellePfannerstill72** [comment]
Did you mean that you leave those 2 or rest of the above beyond them?

**Comment by slimm609** [comment]
The final 2 are not critical,  all others have been updated.   Closing
